### PR TITLE
feat: Sequence integration tests

### PR DESCRIPTION
**Pull request by tonyxiao**
Fixes: #### 

[Describe your change here]

## Test Plan

[Describe test plan here]

## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

**Review by asdfryan — CHANGES_REQUESTED** [review]
I think we should also test `sequenceState`.

Something new that we decided to commit to was to also test invalid request bodies and test that the error responses include relevant context.

File: packages/core/remotes/impl/salesloft/mappers.ts [hunk 1]
```diff
@@ -396,9 +396,7 @@ export const toSalesloftCadenceStepImportParams = (step: SequenceStepCreateParam
           day,
           automated: step.type === 'auto_email',
           automated_settings:
-            step.type === 'auto_email' && delayInMins !== 0
-              ? { send_type: 'after_time_delay', delay_time: delayInMins }
-              : undefined,
+            step.type === 'auto_email' ? { send_type: 'after_time_delay', delay_time: delayInMins } : undefined,
```
Comment (asdfryan) [inline packages/core/remotes/impl/salesloft/mappers.ts:399]: I think you may have to update the existing mapper unit tests

File: apps/api/routes/engagement/v2/sequence.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        {
+          headers: { 'x-provider-name': providerName },
+        }
+      );
+      expect(response.status).toEqual(201);
+      expect(response.data.record?.id).toBeTruthy();
+      addedObjects.push({
+        id: response.data.record?.id as string,
+        providerName,
+        objectName: 'account',
```
Comment (asdfryan) [inline apps/api/routes/engagement/v2/sequence.integration.test.ts:81]: This doesn't look quite right? It should be `sequence` -- and there should probably be additional cleanup code written in `integration-test-environment.js`

Comment (tonyxiao) [reply apps/api/routes/engagement/v2/sequence.integration.test.ts:81]: Thanks for the catch!

**Comment by tonyxiao** [comment]
Done for both

File: apps/api/routes/engagement/v2/sequence.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+      const response = await apiClient.post<CreateSequenceResponse>(
+        '/engagement/v2/sequences',
+        { record: testSequence },
+        {
+          headers: { 'x-provider-name': providerName },
+        }
+      );
+      expect(response.status).toEqual(201);
+      expect(response.data.record?.id).toBeTruthy();
+      addedObjects.push({
```
Comment (asdfryan) [inline apps/api/routes/engagement/v2/sequence.integration.test.ts:115]: Does this get cleaned up at the end even if there are active contacts on it?

File: apps/api/routes/engagement/v2/sequence.integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+const APOLLO_MAILBOX_ID = '651c4cb3520e8800a3808e8a';
+
+describe('sequence', () => {
+  let testSequence: ReturnType<typeof getTestSequence>;
+
+  beforeEach(() => {
+    testSequence = getTestSequence();
+  });
+
+  test.skip(`Error body`, async () => {
```
Comment (asdfryan) [inline apps/api/routes/engagement/v2/sequence.integration.test.ts:93]: are we intentionally skipping?

**Review by asdfryan — APPROVED** [review]

File: apps/api/routes/engagement/v2/sequence.integration.test.ts (hunk 3 above)
Comment (tonyxiao) [reply apps/api/routes/engagement/v2/sequence.integration.test.ts:115]: image
Yes, it appears so.

Comment (tonyxiao) [reply apps/api/routes/engagement/v2/sequence.integration.test.ts:115]: Though delete is only supported in outreach api though.

Comment (tonyxiao) [reply apps/api/routes/engagement/v2/sequence.integration.test.ts:115]: image


And for apollo I don't think there is delete at all, only archive.

**Pull request merged**